### PR TITLE
Add com.giuspen.cherrytree

### PR DESCRIPTION
**Pull request by giuspen**

### CODE REVIEW

File: com.giuspen.CherryTree.json [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+      "sources": [
+        {
+          "type": "archive",
+          "url": "https://www.cairographics.org/releases/cairo-1.16.0.tar.xz",
+          "sha256": "5e7b29b3f113ef870d1e3ecf8adf21f923396401604bda16d44be45e66052331"
+        }
+      ]
+    },
+    {
+      "name": "gtk2",
```
Comment (bilelmoussaoui) [inline com.giuspen.CherryTree.json:37]: Use the one from shared-modules https://github.com/flathub/shared-modules

File: com.giuspen.CherryTree.json [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+      "sources": [
+        {
+          "type": "archive",
+          "url": "https://download.gnome.org/sources/gtksourceview/2.10/gtksourceview-2.10.5.tar.bz2",
+          "sha256": "c585773743b1df8a04b1be7f7d90eecdf22681490d6810be54c81a7ae152191e"
+        }
+      ]
+    },
+    {
+      "name": "python",
```
Comment (bilelmoussaoui) [inline com.giuspen.CherryTree.json:57]: Same as gtk2

File: com.giuspen.CherryTree.json [hunk 3]
```diff
@@ -0,0 +1,145 @@
+{
+  "app-id": "com.giuspen.CherryTree",
+  "runtime": "org.freedesktop.Platform",
+  "runtime-version": "18.08",
```
Comment (bilelmoussaoui) [inline com.giuspen.CherryTree.json:4]: Bump this to 19.08 if possible

**Comment by bilelmoussaoui** [comment]
bot, build com.giuspen.CherryTree

File: com.giuspen.CherryTree.json (hunk 3 above)
Comment (giuspen) [reply com.giuspen.CherryTree.json:4]: ok

File: com.giuspen.CherryTree.json (hunk 1 above)
Comment (giuspen) [reply com.giuspen.CherryTree.json:37]: I don't know how to do that can you suggest the change please?

Comment (giuspen) [reply com.giuspen.CherryTree.json:37]: ok I see there's a README sorry I'll have a look

**Comment by nedrichards** [comment]
The validator can't find the appdata, I assume that needs to be renamed as well since it doesn't have the app-id as the name upstream https://github.com/giuspen/cherrytree/blob/master/linux/cherrytree.appdata.xml looking at that file, you'll fail validation on not having OARS and version tags for sure once it's named correctly. Check https://github.com/flathub/flathub/wiki/AppData-Guidelines for tips.

**Comment by nedrichards** [comment]
bot, build com.giuspen.cherrytree

**Comment by giuspen** [comment]
What changes are you waiting for? The use of shared libraries for gtk and python is done later from my allocated repo as far as I understood, then I applied renaming and OARS. I thought we can give it a go. I have not understood how to run that validator on the mime XML though, not clear from that page AppData-Guidelines.

**Comment by giuspen** [comment]
Cool let's see how it goes

**Comment by giuspen** [comment]
Seems I have to use 0.18.3 after all

**Comment by bilelmoussaoui** [comment]
No you don't, there's intltool in shared-modules which you have to add as well. Using gtk2 and python2 from shared-modules isn't a suggestion but a requirement as it's a burden to maintain, it's already done by someone else, no need to duplicate the effort there.

**Comment by giuspen** [comment]
Ok I understand but I thought the git submodule was to be added later once the build was approved. I'll do and get back thanks.

**Comment by bilelmoussaoui** [comment]
bot, build com.giuspen.cherrytree

File: com.giuspen.cherrytree.json [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+      ]
+    },
+    {
+      "name": "pygtk",
+      "sources": [
+        {
+          "type": "archive",
+          "url": "https://download.gnome.org/sources/pygtk/2.24/pygtk-2.24.0.tar.bz2",
+          "sha256": "cd1c1ea265bd63ff669e92a2d3c2a88eb26bcd9e5363e0f82c896e649f206912"
+        }
```
Comment (bilelmoussaoui) [inline com.giuspen.cherrytree.json:83]: You need to add a new source to pygtk as below
```
                {
                    "type": "shell",
                    "commands": [
                        "cp -p /usr/share/automake-*/config.{sub,guess} ."
                    ]
                }
```

Comment (bilelmoussaoui) [reply com.giuspen.cherrytree.json:83]: This should fix the build in x86_64 at least for now

Comment (giuspen) [reply com.giuspen.cherrytree.json:89]: Thanks!

**Comment by bilelmoussaoui** [comment]
bot, build com.giuspen.cherrytree

**Comment by bilelmoussaoui** [comment]
I looked a bit at the build issues, gtk2 will try to link against a newer pango release which fails currently; you need to embed an older release to fix that. Once that's done, the pygtksourceview fails to build the docs which could be disabled.

I was able to build it successfully with these modifications 
```diff
diff --git a/com.giuspen.cherrytree.json b/com.giuspen.cherrytree.json
index 3e88a48..68f0aa1 100644
--- a/com.giuspen.cherrytree.json
+++ b/com.giuspen.cherrytree.json
@@ -7,8 +7,7 @@
   "rename-desktop-file": "cherrytree.desktop",
   "rename-icon": "cherrytree",
   "finish-args": [
-    "--socket=x11",
-    "--socket=wayland",
+    "--socket=x11",  
     "--filesystem=host"
   ],
   "cleanup": [
@@ -49,9 +48,7 @@
     {
       "name": "pycairo",
       "buildsystem": "meson",
-      "config-opts": [
-        "-Dpython=python"
-      ],
+      "config-opts": ["-Dpython=python"],
       "sources": [
         {
           "type": "archive",
@@ -62,14 +59,22 @@
     },
     {
       "name": "pygobject",
-      "config-opts": [
-        "--disable-introspection"
-      ],
+      "config-opts": ["--disable-introspection"],
+      "sources": [
+        {
+          "type": "archive",
+          "url": "https://download.gnome.org/sources/pygobject/2.28/pygobject-2.28.7.tar.xz",
+          "sha256": "bb9d25a3442ca7511385a7c01b057492095c263784ef31231ffe589d83a96a5a"
+        }
+      ]
+    },
+    {
+      "name": "pango",
       "sources": [
         {
           "type": "archive",
-          "url": "https://download.gnome.org/sources/pygobject/2.28/pygobject-2.28.6.tar.xz",
-          "sha256": "fb8a1d4f665130a125011659bd347c7339c944232163dbb9a34fd0686577adb8"
+          "url": "https://download.gnome.org/sources/pango/1.38/pango-1.38.1.tar.xz",
+          "sha256": "1320569f6c6d75d6b66172b2d28e59c56ee864ee9df202b76799c4506a214eb7"
         }
       ]
     },
@@ -83,14 +88,13 @@
         },
         {
           "type": "shell",
-          "commands": [
-            "cp -p /usr/share/automake-*/config.{sub,guess} ."
-          ]
+          "commands": ["cp -p /usr/share/automake-*/config.{sub,guess} ."]
         }
       ]
     },
     {
       "name": "pygtksourceview",
+      "config-opts": ["--disable-docs"],
       "sources": [
         {
           "type": "archive",

```

**Comment by giuspen** [comment]
Thanks @bilelmoussaoui !

**Comment by bilelmoussaoui** [comment]
bot, build com.giuspen.cherrytree

**Comment by bilelmoussaoui** [comment]
The build fails now as the appdata file seems to be missing 
`stat: cannot stat 'builddir/*/share/app-info/xmls/com.giuspen.cherrytree.xml.gz': No such file or directory` It should be installed under `$prefix/share/metainfo` or `$prefix/share/appdata` and named `com.giuspen.cherrytree.metainfo.xml`

**Comment by bilelmoussaoui** [comment]
for aarch64, you can either disable it https://github.com/flathub/flathub/wiki/App-Maintenance#dont-build-on-arm or add a config.guess to fix the build, see the example in https://github.com/flathub/com.inform7.IDE/blob/master/com.inform7.IDE.yml#L72-L75

**Comment by giuspen** [comment]
Thanks @bilelmoussaoui I hope I got it right now

File: com.giuspen.cherrytree.json [hunk 5]
```diff
@@ -21,6 +21,9 @@
     "/share/doc/",
     "/man/"
   ],
+  "skip-arches": [
```
Comment (bilelmoussaoui) [inline com.giuspen.cherrytree.json:24]: the skip-arches should go in a flathub.json file, to limit which builders to use and not in the manifest itself

**Comment by giuspen** [comment]
Apologies read too fast

**Comment by bilelmoussaoui** [comment]
bot, build com.giuspen.cherrytree

File: flathub.json [hunk 6]
```diff
@@ -0,0 +1,5 @@
+{
+  "skip-arches": [
+    "arm"
```
Comment (bilelmoussaoui) [inline flathub.json:3]:
```suggestion
    "aarch64"
```
arm was building just fine :)

**Comment by bilelmoussaoui** [comment]
The metainfo lacks a release tag https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-releases

**Comment by giuspen** [comment]
Thanks, this may be the good one :)

**Comment by bilelmoussaoui** [comment]
bot, build com.giuspen.cherrytree

**Comment by bilelmoussaoui** [comment]
Looks good to me, have you got a chance to try the latest build ? Otherwise seems ready to go

**Comment by giuspen** [comment]
I rebuilt locally using the exact same scripts and tried that which runs good and has a better gtk theme than before not using the shared gtk2! What comes next?

**Pull request closed without merging**